### PR TITLE
Implement each simulator as a subcommand of yarn sim

### PR DESCRIPTION
**Pull request by djmitche**
This gets nicer --help output and also means each simulator can take its
own parameters on the command line.

This was part of my work on #3360 but now that we have a React viewer I think that's going to take a totally different tack.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
+    const [_, name] = m;
+    const {Simulator, setup} = require(`./sims/${name}`);
+    const subcommand = new commander.Command(name);
+    setup(subcommand);
+    subcommand.action((...args) => {
+      const simOptions = sim.opts();
+
+      const simulator = new Simulator({
+        logging: !simOptions.quiet,
+        commandArgs: args.slice(0, args.length - 2),
```
Comment (milescrabill) [inline main.js]: This is interesting to me, it looks like what gets passed into the action here is a list that includes a `Command` object and a sub-list of variadic arguments. Why are we snipping off the last two elements here?

Essentially, it looks like `commander` already handles grabbing the args for us.

My impression is that this section should look like this:

```
subcommand.action((cmdObj, args) => {
  const simOptions = sim.opts();
  const simulator = new Simulator({
    logging: !simOptions.quiet,
    commandArgs: args,
    commandOptions: cmdObj.opts(),
  });
});
```

**Review by milescrabill — APPROVED** [review]
I checked this out and ran the simple simulation using the new `yarn sim simple` command, everything worked as expected - for example: `yarn sim simple -d 10000`.

I like the way you dynamically create the subcommands so that commander is aware of them, that's very cool.

The one nit I have is the comment about passing arguments to the subcommand, I think that should be fixed before landing this.

File: main.js (hunk 1 above)
Comment (djmitche) [reply main.js]: The last value in `args` is the options object, and the rest are positional things.  So I meant to slice off the last argument, but it looks like I sliced off too much.  I'll test this out with something that takes positional args.

Comment (djmitche) [reply main.js]: Yep, you're right -- I changed it to `- 1` to just slice off the trailing options object.

**Pull request merged**